### DISCUSSION TITLE
Model compile mode. What is good for?

### DISCUSSION THREAD

**Discussion by Centurion-Rome** [vladmandic/sdnext#1678] [Q&A]
Any tips or short info about model compilation? 
![image](https://github.com/vladmandic/automatic/assets/75789136/d1c79c6d-a326-4bf9-a186-c361e87c6dc6)

Eventual gains for Nvidia GPU?

**Comment by vladmandic** [comment]
thats a veeery long and complex story, i suggest reading <https://pytorch.org/get-started/pytorch-2.0/> and <https://pytorch.org/docs/stable/dynamo/index.html> to start with.

**Comment by drax-xard** [comment]
If I understand correctly, once it compiles it's much faster, but you have to recompile (which takes time) every time you change parameters and many standard SD functionalities don't work yet with this option.

I tried it a couple of weeks ago and it was, indeed, much faster but not really usable for my workflow since I like tinkering with all the buttons, nobs and levers to get something close to what I want. And having to wait for recompile was meh.

It may be useful for some cases were all parameters are fixed and you just need high inference speed above else. (Videogames? Real time img2img for video?)

**Reply by vladmandic** [reply]
correct.
